### ISSUE TITLE
Misleading error message, platform mix-up in construct.yaml

### ISSUE THREAD

**Issue by megies** [conda/constructor#83]
When building an installer for `linux-32` on my `linux-64` machine I get a misleading error message:

```
$ constructor --platform=linux-32 1.0.3_py27
platform: linux-32
Fetching package metadata: ......
Solving package specifications: .
Error:  Package missing in current linux-64 channels: 
  - geographiclib
```

I have this in the `constructor.yaml`:

```
specs:
   - [...]
   - geographiclib  [linux-64]
 ```

This could be connected to this issue:

The following tries to include `geographiclib` for both 32/64bit linux:
```
  - geographiclib  [linux-64]
```

And this does not include `geographiclib` for any of 32/64bit linux:
```
  - geographiclib  [linux and not linux-32]
```

**Comment by mbargull** [comment]
TLDR: use `linux64` instead.

The selectors in `construct.yaml` are actual Python expression which get evaluated. Since Python identifiers cannot contain `-` (which is used for subtraction expressions), the corresponding selectors are named differently, i.e., `linux64` and `linux32`, see https://github.com/conda/constructor/blob/e58279ed821b156f83b3adba805016b56daf35bc/constructor/construct.py#L137.
If your selector is `[linux-64]` the expression equals to subtracting `64` from the Boolean variable `linux` (which will be converted to integers `0` or `1`) which evaluates to `-64` or `-63` which are both "true" values, meaning your selector will always be active.

**Comment by megies** [comment]
OK, thanks for the enlightenment, @mbargull. ✨ 

I would strongly suggest then, that this is mentioned in some prominent place (e.g. listing all potential values), e.g. the maxiconda example.
